From 8eb0523a91a94c5c49a9ab7e0c9dac53e3a192ef Mon Sep 17 00:00:00 2001
From: Nicholaus Clark <nicholaus.clark115@gmail.com>
Date: Wed, 9 Aug 2023 10:32:45 -0400
Subject: [PATCH 1/1] Require unistd.h for macosx in libzip

---
 contrib/libzip/premake5.lua | 1 +
 1 file changed, 1 insertion(+)

diff --git a/contrib/libzip/premake5.lua b/contrib/libzip/premake5.lua
index 1532d518..c35592a5 100644
--- a/contrib/libzip/premake5.lua
+++ b/contrib/libzip/premake5.lua
@@ -21,3 +21,4 @@ project "zip-lib"
 
 	filter "system:macosx"
 		defines { "HAVE_SSIZE_T_LIBZIP" }
+		forceincludes { "unistd.h" }
-- 
2.44.2

