From a4fa1da72cc969bb85e575be25acd5aaef2edc49 Mon Sep 17 00:00:00 2001
From: r-a-sattarov <51679282+r-a-sattarov@users.noreply.github.com>
Date: Sun, 8 Dec 2024 22:39:47 +0300
Subject: [PATCH] premake.h - added e2k definition (#2349)

---
 src/host/premake.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/host/premake.h b/src/host/premake.h
index 4e15b863..25ccf33a 100644
--- a/src/host/premake.h
+++ b/src/host/premake.h
@@ -64,6 +64,8 @@
 #define PLATFORM_ARCHITECTURE "RISCV64"
 #elif (defined(__loongarch__) && __loongarch_grlen == 64) || defined(__loongarch64)
 #define PLATFORM_ARCHITECTURE "loongarch64"
+#elif defined(__e2k__)
+#define PLATFORM_ARCHITECTURE "e2k"
 #elif !defined(RC_INVOKED)
 #error Unknown architecture detected
 #endif
-- 
2.45.2

